### ISSUE TITLE
Created receipts saved in server's tmp folder not in CiviCRM temporary folder

### ISSUE THREAD

**Issue by tobiberlin** [systopia/de.systopia.donrec#100]
Hi,

I wondered why the tmp folder of our server became so large and realized that all the created PDF receipts are saved here. I am new to CiviCRM and this module but I did not find a way to change this. As the temporay folder is set in CiviCRM settings I wonder while these files are not saved within this folder?! It also seems that the receipts are never deleted from /tmp folder?!

Best,
Tobias

**Comment by bjendres** [comment]
That's a valid point. 

It was implemented this way, so that people would be able to still access the generated PDF files (for a while) even if e.g. the download failed. As you probably know, if you don't save the downloaded ZIP file, you cannot (easily) recover the original receipt via the UI.

This was under the assumption, that the ``/tmp`` folder is cleared out in regular intervals, which is the case on most of our servers.

We could implement a setting that let's you set the tmp folder yourself, and could offer
* Server tmp
* CiviCRM tmp
* Custom

@tobiberlin What do you think?

**Comment by tobiberlin** [comment]
@bjendress sorry for the late response. I think this would really nice. Esspecially as some users may want to keep the created PDFs for some reasons?! In this way one could define any folder. But to be honest I would not offer the server's tmp folder. To me this is only a very technical space where I wouldn't store temporary files of an application - when there is a local tmp folder for the application.

**Comment by svenseeberg** [comment]
> Especially as some users may want to keep the created PDFs for some reasons?! 

I agree. We are currently facing this issue. We want to archive the actual receipts that have been created. We usually directly send them to the recipients via mail and have no direct option to directly get a copy.

For the time being I will create a script that uploads all created files with a cron job.
